code/game/gamemodes/cult/talisman.dm

```diff
@@ -1,3 +1,585 @@
+		var/datum/action/innate/cult/blood_spell/new_spell = new BS(owner)
+		new_spell.Grant(owner, src)
+		spells += new_spell
+		Positioning()
+		to_chat(owner, "<span class='warning'>Your wounds glows with power, you have prepared a [new_spell.name] invocation!</span>")
+	channeling = FALSE
+
+/datum/action/innate/cult/blood_spell //The next generation of talismans
+	name = "Blood Magic"
+	button_icon_state = "telerune"
+	desc = "Fear the Old Blood."
+	var/charges = 1
+	var/magic_path = null
+	var/obj/item/melee/blood_magic/hand_magic
+	var/datum/action/innate/cult/blood_magic/all_magic
+	var/base_desc //To allow for updating tooltips
+	var/invocation
+	var/health_cost = 0
+
+/datum/action/innate/cult/blood_spell/Grant(mob/living/owner, datum/action/innate/cult/blood_magic/BM)
+	if(health_cost)
+		desc += "<br>Deals <u>[health_cost] damage</u> to your arm per use."
+	base_desc = desc
+	desc += "<br><b><u>Has [charges] use\s remaining</u></b>."
+	all_magic = BM
+	..()
+
+/datum/action/innate/cult/blood_spell/Remove()
+	if(all_magic)
+		all_magic.spells -= src
+	if(hand_magic)
+		qdel(hand_magic)
+		hand_magic = null
+	..()
+
+/datum/action/innate/cult/blood_spell/IsAvailable()
+	var/datum/antagonist/cult/C = owner.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+	if(!C || !charges)
+		return FALSE
+	return ..()
+
+/datum/action/innate/cult/blood_spell/Activate()
+	if(magic_path) //If this spell flows from the hand
+		if(!hand_magic)
+			hand_magic = new magic_path(owner, src)
+			if(!owner.put_in_hands(hand_magic))
+				qdel(hand_magic)
+				hand_magic = null
+				to_chat(owner, "<span class='warning'>You have no empty hand for invoking blood magic!</span>")
+				return
+			to_chat(owner, "<span class='notice'>Your old wounds glow again as you invoke the [name].</span>")
+			return
+		if(hand_magic)
+			qdel(hand_magic)
+			hand_magic = null
+			to_chat(owner, "<span class='warning'>You snuff out the spell with your hand, saving its power for another time.</span>")
+
+
+//Cult Blood Spells
+/datum/action/innate/cult/blood_spell/stun
+	name = "Stun"
+	desc = "A potent spell that will stun and mute victims upon contact."
+	button_icon_state = "hand"
+	magic_path = "/obj/item/melee/blood_magic/stun"
+	health_cost = 10
+
+/datum/action/innate/cult/blood_spell/teleport
+	name = "Teleport"
+	desc = "A useful spell that teleport cultists to a chosen destination on contact."
+	button_icon_state = "tele"
+	magic_path = "/obj/item/melee/blood_magic/teleport"
+	health_cost = 5
+
+/datum/action/innate/cult/blood_spell/emp
+	name = "Electromagnetic Pulse"
+	desc = "A large spell that immediately disables all electronics in the area."
+	button_icon_state = "emp"
+	health_cost = 5
+
+/datum/action/innate/cult/blood_spell/emp/Activate()
+	owner.visible_message("<span class='warning'>[owner]'s hand flashes a bright blue!</span>", \
+						 "<span class='cultitalic'>You speak the cursed words, emitting an EMP blast from your hand.</span>")
+	empulse(src, 4, 8)
```

Comment:
i don't think you should be empulsing a /datum.

Reply:
Hmm it worked in testing but that's pretty much copypasted EMP talisman code. How is it OP, its an EMP talisman... which was extremely underutilized both as a talisman and a rune?

Reply:
eh, even if you strip them this means you can use it, no?
the entire thing about blood magic means you'd need to keep cultists cuffed lest they channel their magic and ggnore you after you put them in a cell and an AOE effect to boot is kinda op to me but whatever i won't scream and cry until a testmerge happens.

Reply:
should be owner not src